system/system.go

```diff
@@ -59,9 +61,35 @@ type Cmd struct {
 	*exec.Cmd
 }
 
+func (c *Cmd) Run() error {
+	err := c.Cmd.Run()
+	if logger, ok := c.Stdout.(Flusher); ok {
+		logger.Flush()
+	}
+	if logger, ok := c.Stderr.(Flusher); ok {
+		logger.Flush()
+	}
+	return err
+}
+
+func (c *Cmd) Wait() error {
+	err := c.Cmd.Wait()
+	if logger, ok := c.Stdout.(Flusher); ok {
+		logger.Flush()
+	}
+	if logger, ok := c.Stderr.(Flusher); ok {
+		logger.Flush()
+	}
+	return err
+}
+
 func (c *Cmd) Output() ([]byte, error) {
 	c.Stdout = nil
-	return c.Cmd.Output()
+	b, err := c.Cmd.Output()
+	if logger, ok := c.Stderr.(Flusher); ok {
+		logger.Flush()
+	}
+	return b, err
 }
 
 func (c *Cmd) CombinedOutput() ([]byte, error) {
```

Comment:
`Flush()` should be called in here as well, or?

Reply:
No, it should not :)
If you think about it you know why ;)

Reply:
Haha, you're right, I see it now!